src/rfc3986/misc.py

```diff
@@ -128,4 +136,4 @@ def merge_paths(base_uri, relative_path):
         return path[:index] + "/" + relative_path
 
 
-UseExisting = object()
+UseExisting: t.Any = object()
```

Comment:
So this is meant to be a sentinel value. I'm not sure we need to go as far as they did in urllib3, but a similar sentinel value exists there:

https://github.com/urllib3/urllib3/blob/f5196167eff3c5e4d696ae9d4c2917776a092bf3/src/urllib3/util/timeout.py#L8-L22

Reply:
At the very least it should be `t.Final[t.Any]` but I've never seen `t.Any` as a type to `t.Final` so not sure how well that will work

Reply:
I prefer the `sentinel: Any = object()` idiom because I'm lazy and don't want to include the sentinel type in annotations everywhere the sentinel is used; however, `UseExisting` is referenced only 22 times in 3 methods and the Union including the hypothetical base enum type can be aliased as your link shows, so it wouldn't be a terrible burden in theory.

Reply:
> At the very least it should be `t.Final[t.Any]` but I've never seen `t.Any` as a type to `t.Final` so not sure how well that will work.

Oh, this message didn't load for me for some reason. Yeah, `Final` seems fine to warn users off from reassigning that variable, assuming they're using a type checker.